sdk/storage/azure-storage-file-share/azure/storage/fileshare/_share_client.py

```diff
@@ -391,10 +391,19 @@ def delete_share(
         later deleted during garbage collection.
 
         :param bool delete_snapshots:
-            Indicates if snapshots are to be deleted.
+            Indicates if snapshots are to be deleted. This parameter is being deprecated.
+        :keyword include_snapshots:
+            Specifies which snapshots to delete.
+            Possible values: 'include' which includes non-leased snapshots,
+            and 'include-with-leased' which includes all snapshots.
```

Comment:
I feel we can add delete_snapshots_including_leased as a bool, instead of having include_snapshots and delete_snapshots both

Reply:
I'd rather not do that. This enum can expand over time. We don't want to have explosion of bool flags and APIs.

Reply:
I'm good with it. I just feel delete_snapshots and include_snapshots has overlapping function, it could cause confusion to customers. While it makes sense that later we have more options to include.

Reply:
Why can we not just use the existing parameter for this behaviour? Where it's value could be the existing `True`/`False`, or an enum with more fine-grained behaviour options?
We should make sure that setting this value to `True` maps to the most sensible, non-breaking behaviour.

Reply:
@annatisch that's an excellent idea!! not sure why I didn't think of that :sweat_smile:, as for the most sensible non-breaking behavior for `True` I actually think it should delete **all** snapshots, given that leased share snapshots are a recent addition. I feel that customers who are using this param are just expecting to delete all their snapshots